app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -528,42 +551,38 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
             Q.sortBy('latest', Q.desc),
         ).fetch()) as PostsInChannelModel[];
 
-        if (chunks.length === 0) {
-            // No chunks found, previous posts in this block not found
+        if (!chunks.length) {
+            // Create a new chunk in case somehow the chunks got deleted for this channel
+            const latest = lastPost.create_at;
+            return this._createPostsInChannelRecord(channelId, earliest, latest, prepareRecordsOnly);
+        }
+
+        const targetChunk = chunks[0];
+        if (targetChunk.earliest <= earliest) {
             return [];
         }
 
-        let targetChunk = chunks[0];
-        if (targetChunk) {
-            if (targetChunk.earliest <= earliest) {
-                return [];
-            }
+        const models = [];
 
-            // If the chunk was found, Update the chunk and return
-            if (prepareRecordsOnly) {
-                targetChunk.prepareUpdate((record) => {
-                    record.earliest = Math.min(record.earliest, earliest);
-                });
-                return [targetChunk];
-            }
+        // If the chunk was found, Update the chunk and return
+        models.push(targetChunk.prepareUpdate((record) => {
+            record.earliest = Math.min(record.earliest, earliest);
+        }));
 
-            targetChunk = await this.database.write(async () => {
-                return targetChunk!.update((record) => {
-                    record.earliest = Math.min(record.earliest, earliest);
-                });
-            });
+        models.push(...(await this._mergePostInChannelChunks(targetChunk, chunks, prepareRecordsOnly)));
```

Comment:
Getting posts before may intersect with other previous chunks, so we need to merge them.